packages/element/src/utils.ts

```diff
@@ -4,13 +4,17 @@
  * @param {*} element WP element to check.
  * @return {boolean} True when an element is considered empty.
  */
-export const isEmptyElement = ( element ) => {
+export const isEmptyElement = ( element: unknown ): boolean => {
 	if ( typeof element === 'number' ) {
 		return false;
 	}
 
-	if ( typeof element?.valueOf() === 'string' || Array.isArray( element ) ) {
-		return ! element.length;
+	if (
+		typeof ( element as { valueOf: () => unknown } )?.valueOf() ===
+			'string' ||
+		Array.isArray( element )
+	) {
+		return ! ( element as { length: number } ).length;
 	}
```

Comment:
We don't need the first assertion here

```suggestion
	if ( typeof element?.valueOf() === 'string' || Array.isArray( element ) ) {
		return ! ( element as { length: number } ).length;
	}
```